thunder/tests/test_networks.py

```diff
@@ -404,23 +407,41 @@ def test_thunderfx_mistral_nemo_small():
     assert mdl._backend.subgraph_infos, "Should have at least 1 subgraph"
 
 
+def _get_model_config_pairs():
+
+    def phi3():
+        from transformers.models.phi3 import Phi3ForCausalLM, Phi3Config
+
+        return Phi3ForCausalLM, Phi3Config
+
+    def qwen2():
+        from transformers.models.qwen2 import Qwen2ForCausalLM, Qwen2Config
+
+        return Qwen2ForCausalLM, Qwen2Config
+
+    return [phi3(), qwen2()]
+
+
 @thunder.tests.framework.requiresCUDA
-@pytest.mark.parametrize("model_id", ["Qwen/Qwen2.5-7B-Instruct", "microsoft/Phi-3-mini-128k-instruct"])
-def test_hf_for_nemo(model_id):
+@pytest.mark.parametrize("model_cls, config_cls", _get_model_config_pairs())
```

Comment:
But now you still need to have transformers to run the file.
Maybe you can return the functions and only call it in the test itself?